libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -718,32 +745,26 @@ public void appendMediaFiles(Map<String, MediaFile> mediaList) {
             mediaUrl = (String) mediaUrls[0];
         }
 
-        if (URLUtil.isNetworkUrl(mediaUrl)) {
-            for (Map.Entry<String, MediaFile> mediaEntry : mediaList.entrySet()) {
-                rnMediaList.add(
-                        new Media(
-                                Integer.valueOf(mediaEntry.getValue().getMediaId()),
-                                mediaEntry.getKey(),
-                                mediaEntry.getValue().getMimeType()
-                        )
-                );
-            }
-            getGutenbergContainerFragment().appendMediaFiles(rnMediaList);
-        } else {
-            for (Map.Entry<String, MediaFile> mediaEntry : mediaList.entrySet()) {
-                rnMediaList.add(
-                        new Media(
-                                mediaEntry.getValue().getId(),
-                                "file://" + mediaEntry.getKey(),
-                                mediaEntry.getValue().getMimeType()
-                        )
-                );
-            }
-            getGutenbergContainerFragment().appendUploadMediaFiles(rnMediaList);
+        boolean isNetworkUrl = URLUtil.isNetworkUrl(mediaUrl);
+        if (!isNetworkUrl) {
             for (Media media : rnMediaList) {
                 mUploadingMediaProgressMax.put(String.valueOf(media.getId()), 0f);
             }
         }
+
+        for (Map.Entry<String, MediaFile> mediaEntry : mediaList.entrySet()) {
+            rnMediaList.add(
+                    new Media(
+                            isNetworkUrl
+                                    ? Integer.valueOf(mediaEntry.getValue().getMediaId())
```

Comment:
No changes needed here. I just wanted to comment that this line of code make me miss Kotlin's null safety https://kotlinlang.org/docs/reference/null-safety.html. Here I needed to verify that it is not possible to add an entry with a null value to mediaList , while Kotlin would allow you to specify that a mediaEntry has MediaFile values that can not be null. If we ever see NPE crashes in the app, it would be worth thinking about translating some of these classes to Kotlin, but in the mean time, it is probably not a big deal.

Reply:
Thanks for the comment @cameronvoell!
I would like to see us moving forward with the Kotlin, so probably we need to start pushing its usage in our project.